polygeist/tools/cgeist/Test/Verification/sycl/sycl_extern_func.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) Intel
+
+//===--- sycl_extern_func.cpp ---------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// RUN: sycl-clang.py %s -S | FileCheck %s --check-prefix=CHECK-MLIR
+// RUN: sycl-clang.py %s -S -emit-llvm | FileCheck %s --check-prefix=CHECK-LLVM
+
+#include <sycl/sycl.hpp>
+
+// CHECK-MLIR-LABEL: func.func @cons_5() attributes {llvm.cconv = #llvm.cconv<spir_funccc>, llvm.linkage = #llvm.linkage<external>, passthrough = ["norecurse", "nounwind", "convergent", "mustprogress"]} {
+// CHECK-MLIR:           sycl.constructor(%{{.*}})
+// CHECK-MLIR-NEXT:      return
+
+// CHECK-LLVM-LABEL: define spir_func void @cons_5() #0 {
+// CHECK-LLVM-NEXT:  %{{.*}} = alloca %"class.cl::sycl::accessor.1", i64 ptrtoint (%"class.cl::sycl::accessor.1"* getelementptr (%"class.cl::sycl::accessor.1", %"class.cl::sycl::accessor.1"* null, i32 1) to i64), align 8
+// CHECK-LLVM-NEXT:  call void @_ZN4sycl3_V18accessorIiLi1ELNS0_6access4modeE1025ELNS2_6targetE2014ELNS2_11placeholderE0ENS0_3ext6oneapi22accessor_property_listIJEEEEC1Ev(%"class.cl::sycl::accessor.1"* %1, %"class.cl::sycl::accessor.1"* %1, i64 0, i64 1, i64 1)
+// CHECK-LLVM-NEXT:  ret void
+
+
```

Comment:
[SUGGESTION]: Add CHECKS to ensure the kernel is generated and contains a call to cons_5 ?

Reply:
Will add. Thanks!